frame/session/src/lib.rs

```diff
@@ -532,9 +532,9 @@ pub mod pallet {
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event {
-		/// New session has happened. Note that the argument is the \[session_index\], not the
+		/// New session has happened. Note that the argument is the session_index, not the
 		/// block number as the type might suggest.
-		NewSession(SessionIndex),
+		NewSession { index: SessionIndex },
```

Comment:
Probably better to just name it as `session_index`:

```suggestion
		NewSession { session_index: SessionIndex },
```